File: src/learn-topic-wise/07-linked-List/utils/generate_linked_list.py
from typing import List


class Utils:
    class ListNode:
        def __init__(self, x):
            self.val = x
            self.next = None

    def Generate(self, arr: List[int]) -> ListNode:
        head = self.ListNode(arr[0])
        next = head

        for val in arr[1:]:
            next.next = self.ListNode(val)
            next = next.next

        return head

File: src/learn-topic-wise/07-linked-List/utils/generate-linked-list.ts
// Definition for singly-linked list.
export class ListNode {
  val: number;
  next: ListNode | null;
  constructor(val?: number, next?: ListNode | null) {
    this.val = val === undefined ? 0 : val;
    this.next = next === undefined ? null : next;
  }
}

export class LinkedList {
  #internalLinkedList: ListNode | null;
  get singlyList() {
    return this.#internalLinkedList;
  }
  constructor(arr: number[]) {
    if (arr.length === 0) this.#internalLinkedList = null;

    this.#internalLinkedList = new ListNode(arr[0]);

    this.#buildList(arr);
  }
  #buildList(arr: number[]) {
    let cur = this.#internalLinkedList;
    let i = 1;
    while (cur) {
      const nextArrItem = arr[i];
      if (nextArrItem) {
        cur.next = new ListNode(nextArrItem, null);
      }

      cur = cur.next;
      i++;
    }
  }
}

var myLinkedList = new LinkedList([1]);
console.log(myLinkedList.singlyList);
